components/support/interrupt/src/lib.rs

```diff
@@ -0,0 +1,46 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
```

Comment:
Add these to follow other crates (unfortunately there's no way to configure this for the whole workspace
```
#![allow(unknown_lints)]
#![warn(rust_2018_idioms)]
```
(Note: The second was added today, if you get merge conflicts for the lint changes, you can resolve in favor of yours, and run `cargo fix --edition-idioms` to resolve for the most part)